src/coreclr/src/jit/optimizer.cpp

```diff
@@ -9249,7 +9249,7 @@ void Compiler::optRemoveRedundantZeroInits()
             Statement* next = stmt->GetNextStmt();
             for (GenTree* tree = stmt->GetTreeList(); tree != nullptr; tree = tree->gtNext)
             {
-                if (((tree->gtFlags & GTF_CALL) != 0) && (!tree->IsCall() || !tree->AsCall()->IsSuppressGCTransition()))
+                if (((tree->gtFlags & GTF_CALL) != 0))
                 {
                     hasGCSafePoint = true;
```

Comment:
You have made this optimization more conservative.
This local 'hasGCSafePoint ' probably should now be renamed 'hasCall' .

There is also a method called `Compiler::IsGcSafePoint` that can be used to determine if a call is a GC safe point.
It is currently only used by fgMorph to set the BBF_GC_SAFE_POINT flag on a BasicBlock

Reply:
This is intentional. This code needs to know if the node is a potential GC-safe point. Since now we insert GC Polls after this optimization, we have to treat calls to methods with `SuppressGCTransitionAttribute` as potential GC-safe points.

`Compiler::IsGcSafePoint` will return true if the node is definitely a GC-safe point and will return false if the node may or may not be a GC-safe point so it shouldn't be used here.